src/Talis/Persona/Client/Login.php

```diff
@@ -61,9 +61,19 @@ public function requireAuth($provider, $appId, $appSecret, $redirectUri = '')
      */
     public function validateAuth()
     {
+        if (isset($_POST['persona:signature']))
+        {
+            $payloadSignature = $_POST['persona:signature'];
+        } else {
+            $this->getLogger()->error("Signature not set");
+            throw new \Exception('Signature not set');
```

Comment:
OK, cool, will leave it as is then. Deciding between status codes always gets me by the way, can never make my mind up which is better!